test/common.py

```diff
@@ -2049,23 +2049,32 @@ class TestServerHandler(SimpleHTTPRequestHandler):
     # Request header handler for default do_GET() path in
     # SimpleHTTPRequestHandler.do_GET(self) below.
     def send_head(self):
-      if self.path.endswith('.js'):
-        path = self.translate_path(self.path)
-        try:
-          f = open(path, 'rb')
-        except IOError:
-          self.send_error(404, "File not found: " + path)
-          return None
-        self.send_response(200)
-        self.send_header('Content-type', 'application/javascript')
-        self.send_header('Connection', 'close')
+      path = self.translate_path(self.path)
+      try:
+        fsize = os.path.getsize(path)
+        f = open(path, 'rb')
```

Comment:
It looks like `f` is not closed in the non-Range case below.

Is it even needed in that path?   Maybe this whole try/catch can go inside the `Range` check?

Reply:
https://github.com/python/cpython/blob/de84531a3ec5f1f4357983f2e95e0c614b2e744e/Lib/http/server.py#L683C1-L687C22 The HTTP server does close it; send_head should return an open file for do_GET to do the right thing.